sql/select.go

```diff
@@ -405,6 +411,22 @@ func (v *indexInfo) makeConstraints(exprs []parser.Exprs) {
 						constraint.start = c
 					}
 				case parser.In:
+					// Only allow the IN constraint if the previous constraints are all
+					// EQ. This is necessary to prevent overlapping spans from being
+					// generated. Consider the constraints [a >= 1, a <= 2, b IN (1,
+					// 2)]. This would turn into the spans /1/1-/3/2 and /1/2-/3/3.
+					ok := true
+					for _, c := range v.constraints {
+						if c.start == c.end && c.start.Operator == parser.EQ {
```

Comment:
how about using the ! of this boolean expression and setting ok to false and breaking. Or set ok to the expression and break once ok is false.